src/index.js

```diff
@@ -133,64 +132,66 @@ class Plugin {
 
 		if(config.topics) {
 			Object.keys(config.topics).forEach((key) => {
-				const topicName = config.topics[key];
-				if(topicName) {
-					const cfRef = `AwsAlerts${_.upperFirst(key)}`
-					alertTopics[key] = cfRef;
-					
-					this.addCfResources({
-						[cfRef]: this.getCfSnsTopic(topicName),
-					});
+				const topic = config.topics[key];
+				if (topic) {
+					if (topic.indexOf('arn:') === 0) {
+						alertTopics[key] = topic;
+					} else {
+						const cfRef = `AwsAlerts${_.upperFirst(key)}`;
+						alertTopics[key] = { Ref: cfRef };
+
+						this.addCfResources({
+							[cfRef]: this.getCfSnsTopic(topic),
+						});
+					}
 				}
 			});
 		}
 
 		return alertTopics;
 	}
 
-        getLogMetricCF(alarm, functionName, normalizedFunctionName){
-            var output = {}
-            if (alarm.pattern){
-                //add custom log metric
-                const logMetricCFRef = this.naming.getLogMetricCFRef(normalizedFunctionName,alarm.name)
-                const CFLogName = this.serverless.providers.aws.naming.getLogGroupLogicalId(functionName);
-                output[logMetricCFRef] = {
-                    Type: "AWS::Logs::MetricFilter",
-                    DependsOn: CFLogName,
-                    Properties: {
-                        FilterPattern: alarm.pattern,
-                        LogGroupName: this.serverless.providers.aws.naming.getLogGroupName(this.serverless.service.getFunction(functionName).name),
-                        MetricTransformations:[{
-                            MetricValue: 1,
-                            MetricNamespace:this.serverless.service.service + '_' + alarm.namespace,
-                            MetricName: alarm.metric + normalizedFunctionName
-                        }]
-                    }
-                }    
-
-            }else{
-                // not a custom log metric, nothing to add
-            }
-            return output
-        }
+	getLogMetricCF(alarm, functionName, normalizedFunctionName){
+		var output = {};
+		if (alarm.pattern) {
+      //add custom log metric
+			const logMetricCFRef = this.naming.getLogMetricCFRef(normalizedFunctionName,alarm.name);
+			const CFLogName = this.serverless.providers.aws.naming.getLogGroupLogicalId(functionName);
+			output[logMetricCFRef] = {
+				Type: "AWS::Logs::MetricFilter",
+				DependsOn: CFLogName,
+				Properties: {
+					FilterPattern: alarm.pattern,
+					LogGroupName: this.serverless.providers.aws.naming.getLogGroupName(this.serverless.service.getFunction(functionName).name),
```

Comment:
I think maybe use `serverless.getProvider('aws')` instead (more future-proof if they preserve this getter but don't keep .providers objecT)